### PR TITLE
[wptrunner] Implement `--retry-unexpected`

### PR DESCRIPTION
**Pull request by jonathan-j-lee**
This PR implements: https://github.com/web-platform-tests/rfcs/blob/master/rfcs/retry_unexpected.md

It appears `wptrunner.wptrunner` has no automated tests, so I tested this change manually by making `xhr/abort-after-stop.window.js` a randomly passing test:

```js
test(t => {
  assert_less_than(Math.random(), 0.2, "this test flakes");
});
```

Then trying (with `chrome`):
```
./wpt run ... xhr/abort-after-stop.window.js badging/badge-success.https.html --retry-unexpected=20 --repeat=2 --rerun=5
./wpt run ... xhr/abort-after-stop.window.js badging/badge-success.https.html --retry-unexpected=20 --no-fail-on-unexpected
```

* The first command probably retries only `xhr/abort-after-stop.window.js`, but eventually succeeds.
* The second command retries no tests.

### CODE REVIEW

File: tools/wptrunner/wptrunner/testrunner.py [hunk 1]
```diff
@@ -899,12 +903,11 @@ def __exit__(self, exc_type, exc_val, exc_tb):
     def run(self, test_type, tests):
         """Start all managers in the group"""
         self.logger.debug("Using %i processes" % self.size)
-        type_tests = tests[test_type]
-        if not type_tests:
+        if not tests:
```
Comment (WeizhongX) [inline tools/wptrunner/wptrunner/testrunner.py:906]: Is this correct? there can be different type of test in tests, right?

Also looks like this is a clean up. I suggest we do that in a different PR, to speed up the review process.

**Review by WeizhongX — APPROVED** [review]

**Comment by WeizhongX** [comment]
jonathan-j-lee@, this looks good to me, but there are failures on Azure Pipeline.

**Comment by jonathan-j-lee** [comment]
Looked unrelated, which I filed #35974 for.

Also, I realized `retry_unexpected_tests(...)` can be called even when not enabled. Currently, this degrades into a no-op, but that might not hold in the future. Will make a change.
